flow/layers/layer.h

```diff
@@ -261,6 +262,9 @@ class Layer {
 #ifdef FLUTTER_ENABLE_DIFF_CONTEXT
 
   virtual const PictureLayer* as_picture_layer() const { return nullptr; }
+  virtual const DisplayListLayer* as_display_list_layer() const {
+    return nullptr;
+  }
```

Comment:
I know I'm late to the party here, but rather than continuing to add to these, can we just create a virtual method that gives the layer an opportunity to virtually do what it needs to do with the other layer? It looks like we're more or less doing that with the DiffContext object in some places, but that's failing to capture some significance somehow.

Reply:
This would be for @knopp to decide and would apply to the overall implementation of the DiffContext (see https://github.com/flutter/engine/pull/21824)